File: suites/03_data_structures/comparisons/hashmap_operations.go
package main
import "fmt"
func hashmapProbeCost(slots, rounds int64) int64 {
	var total int64
	for r := int64(0); r < rounds; r++ {
		key := (r * 2654435761) % 2147483647
		total += ((key % slots) * 37) % 4099
	}
	return total
}
func main() { fmt.Println(hashmapProbeCost(65536, 5000000)) }

File: suites/03_data_structures/comparisons/hashmap_operations.py
def hashmap_probe_cost(slots: int, rounds: int) -> int:
    total = 0
    for r in range(rounds):
        key = (r * 2654435761) % 2147483647
        total += ((key % slots) * 37) % 4099
    return total
if __name__ == "__main__":
    print(hashmap_probe_cost(65536, 5000000))
